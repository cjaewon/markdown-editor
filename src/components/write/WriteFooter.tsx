import * as React from 'react';
import styled from 'styled-components';
import Button from '../common/Button';
import { MdArrowBack } from 'react-icons/md';
import media from '../../lib/styles/media';


const WriteFooterBlock = styled.div`
  padding-left: 1rem;
  padding-right: 1rem;
  height: 4rem;
  width: 100%;
  box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.1);
  background: rgba(255, 255, 255, 0.85);
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const Group = styled.div`
  justify-content: flex-end;
  align-items: center;
`;

const StyledButton = styled(Button)`
  height: 2.5rem;
  font-size: 1.125rem;
  & + & {
    margin-left: 0.75rem;
  }
`;

const BackButton = styled.button`
  height: 2.5rem;
  padding: 0.5rem 1rem;
  align-items: center;
  background: none;
  border-radius: 4px;
  cursor: pointer;
  border: none;
  display: flex;
  outline: none;
  &:hover,
  &:focus {
    background: rgba(0, 0, 0, 0.05);
  }
  svg {
    font-size: 1.25rem;
    margin-right: 0.5rem;
  }
  span {
    font-size: 1.125rem;
    ${media.xsmall} {
      display: none;
    }
  }
`;

export interface WriteFooterProps {
  onTempSave: (notify?: boolean) => void;
  onPublish: () => void;
  onGoBack: () => void;
  edit: boolean;
}

const WriteFooter: React.FC<WriteFooterProps> = ({
  onGoBack,
  onTempSave,
  onPublish,
  edit,
}) => {
  return (
    <WriteFooterBlock>
      <BackButton onClick={onGoBack}>
        <MdArrowBack />
        <span>나가기</span>
      </BackButton>
      <Group>
        <StyledButton inline color="lightGray" onClick={() => onTempSave(true)}>
          임시저장
        </StyledButton>
        <StyledButton inline color="teal" onClick={onPublish}>
          {edit ? '수정하기' : '출간하기'}
        </StyledButton>
      </Group>
    </WriteFooterBlock>
  );
};

export default WriteFooter;
