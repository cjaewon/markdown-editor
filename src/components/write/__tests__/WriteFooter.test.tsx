import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import WriteFooter, { WriteFooterProps } from '../WriteFooter';

describe('WriteFooter', () => {
  const setup = (props: Partial<WriteFooterProps> = {}) => {
    const initialProps: WriteFooterProps = {
      edit: false,
      onGoBack: () => {},
      onPublish: () => {},
      onTempSave: () => {},
    };
    const utils = render(<WriteFooter {...initialProps} {...props} />);
    const buttons = {
      tempSave: utils.getByText('임시저장'),
      publish: utils.getByText(/(출간|수정)하기/),
      goBack: utils.getByText('나가기'),
    };
    return {
      ...utils,
      buttons,
    };
  };
  it('renders properly', () => {
    setup();
  });
  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('buttons are working properly', () => {
    const onPublish = jest.fn();
    const onTempSave = jest.fn();
    const onGoBack = jest.fn();
    const utils = setup({ onPublish, onTempSave, onGoBack });
    const { tempSave, publish, goBack } = utils.buttons;
    fireEvent.click(tempSave);
    expect(onTempSave).toBeCalled();
    fireEvent.click(publish);
    expect(onPublish).toBeCalled();
    fireEvent.click(goBack);
    expect(onGoBack).toBeCalled();
  });
  it('changes button text when edit is true', () => {
    const { buttons } = setup({
      edit: true,
    });
    expect(buttons.publish.textContent).toBe('수정하기');
  });
});
