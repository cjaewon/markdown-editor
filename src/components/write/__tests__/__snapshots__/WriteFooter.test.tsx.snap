// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WriteFooter matches snapshot 1`] = `
<div>
  <div
    class="sc-bwzfXH prIea"
  >
    <button
      class="sc-ifAKCX ehUtwD"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
      <span>
        나가기
      </span>
    </button>
    <div
      class="sc-htpNat SXVIf"
    >
      <button
        class="sc-bdVaJa eORyJT sc-bxivhb eFySeT"
        color="lightGray"
      >
        임시저장
      </button>
      <button
        class="sc-bdVaJa czPBkf sc-bxivhb eFySeT"
        color="teal"
      >
        출간하기
      </button>
    </div>
  </div>
</div>
`;
